--- PDP11/pdp11_defs.h.orig	Wed May  7 21:04:22 2008
+++ PDP11/pdp11_defs.h	Wed May  7 21:16:51 2008
@@ -470,6 +470,7 @@
 #define DZ_LINES        8                               /* lines per DZ mux */
 #define VH_MUXES        4                               /* max # of VH muxes */
 #define TTX_LINES       16                              /* max # of KL11/DL11's */
+#define DCX_LINES       16                              /* max # of DC11's */
 #define MT_MAXFR        (1 << 16)                       /* magtape max rec */
 #define AUTO_LNT        34                              /* autoconfig ranks */
 #define DIB_MAX         100                             /* max DIBs */
@@ -555,6 +556,8 @@
 #define IOLN_XU         010
 #define IOBA_TTIX       (IOPAGEBASE + 016500)           /* extra KL11/DL11 */
 #define IOLN_TTIX       (TTX_LINES * 010)
+#define IOBA_DCIX       (IOPAGEBASE + 014000)           /* extra DC11 */
+#define IOLN_DCIX       (DCX_LINES * 010)
 #define IOBA_RP         (IOPAGEBASE + 016700)           /* RP/RM */
 #define IOLN_RP         054
 #define IOBA_CR         (IOPAGEBASE + 017160)           /* CD/CR/CM */
@@ -648,6 +653,8 @@
 #define INT_V_CR        7
 #define INT_V_TTIX      8
 #define INT_V_TTOX      9
+#define INT_V_DCIX      8
+#define INT_V_DCOX      9
 #define INT_V_PIR4      10
 
 #define INT_V_PIR3      0                               /* BR3 */
@@ -687,6 +694,8 @@
 #define INT_CR          (1u << INT_V_CR)
 #define INT_TTIX        (1u << INT_V_TTIX)
 #define INT_TTOX        (1u << INT_V_TTOX)
+#define INT_DCIX        (1u << INT_V_DCIX)
+#define INT_DCOX        (1u << INT_V_DCOX)
 #define INT_PIR4        (1u << INT_V_PIR4)
 #define INT_PIR3        (1u << INT_V_PIR3)
 #define INT_PIR2        (1u << INT_V_PIR2)
@@ -722,6 +731,8 @@
 #define IPL_CR          4
 #define IPL_TTIX        4
 #define IPL_TTOX        4
+#define IPL_DCIX        4
+#define IPL_DCOX        4
 
 #define IPL_PIR7        7
 #define IPL_PIR6        6
@@ -761,6 +772,8 @@
 #define VEC_RY          0264
 #define VEC_TTIX        0300
 #define VEC_TTOX        0304
+#define VEC_DCIX        0300
+#define VEC_DCOX        0304
 #define VEC_DZRX        0300
 #define VEC_DZTX        0304
 #define VEC_VHRX        0310
--- PDP11/pdp11_sys.c.orig	Fri Dec 29 12:09:20 2006
+++ PDP11/pdp11_sys.c	Wed May  7 21:35:43 2008
@@ -71,6 +71,8 @@
 extern DEVICE pclk_dev;
 extern DEVICE ttix_dev;
 extern DEVICE ttox_dev;
+extern DEVICE dcix_dev;
+extern DEVICE dcox_dev;
 extern DEVICE dz_dev;
 extern DEVICE vh_dev;
 extern DEVICE dt_dev;
@@ -126,6 +128,8 @@
     &lpt_dev,
     &ttix_dev,
     &ttox_dev,
+    &dcix_dev,
+    &dcox_dev,
     &dz_dev,
     &vh_dev,
     &rf_dev,
--- makefile.orig	Wed May  7 21:37:54 2008
+++ makefile	Wed May  7 21:32:45 2008
@@ -85,7 +85,7 @@
 	${PDP11D}/pdp11_xq.c ${PDP11D}/pdp11_xu.c ${PDP11D}/pdp11_vh.c \
 	${PDP11D}/pdp11_rh.c ${PDP11D}/pdp11_tu.c ${PDP11D}/pdp11_cpumod.c \
 	${PDP11D}/pdp11_cr.c ${PDP11D}/pdp11_rf.c ${PDP11D}/pdp11_dl.c \
-	${PDP11D}/pdp11_ta.c ${PDP11D}/pdp11_ke.c
+	${PDP11D}/pdp11_ta.c ${PDP11D}/pdp11_ke.c ${PDP11D}/pdp11_dc.c
 PDP11_OPT = -DVM_PDP11 -I ${PDP11D} ${NETWORK_OPT}
 
 
