--- makefile.orig	2007-02-06 19:24:54.000000000 -1000
+++ makefile	2008-05-02 07:09:37.000000000 -1000
@@ -85,7 +85,7 @@
 	${PDP11D}/pdp11_xq.c ${PDP11D}/pdp11_xu.c ${PDP11D}/pdp11_vh.c \
 	${PDP11D}/pdp11_rh.c ${PDP11D}/pdp11_tu.c ${PDP11D}/pdp11_cpumod.c \
 	${PDP11D}/pdp11_cr.c ${PDP11D}/pdp11_rf.c ${PDP11D}/pdp11_dl.c \
-	${PDP11D}/pdp11_ta.c
+	${PDP11D}/pdp11_ta.c ${PDP11D}/pdp11_ke.c
 PDP11_OPT = -DVM_PDP11 -I ${PDP11D} ${NETWORK_OPT}
 
--- PDP11/pdp11_defs.h.~1~	2006-07-06 11:33:30.000000000 -0400
+++ PDP11/pdp11_defs.h	2008-05-01 11:12:57.000000000 -0400
@@ -561,6 +561,8 @@
 #define IOLN_RX         004
 #define IOBA_RY         (IOPAGEBASE + 017170)           /* RY11 */
 #define IOLN_RY         004
+#define IOBA_KE         (IOPAGEBASE + 017300)           /* KE11 */
+#define IOLN_KE         020
 #define IOBA_TC         (IOPAGEBASE + 017340)           /* TC11 */
 #define IOLN_TC         012
 #define IOBA_RK         (IOPAGEBASE + 017400)           /* RK11 */
--- PDP11/pdp11_sys.c.~1~	2006-07-14 15:20:48.000000000 -0400
+++ PDP11/pdp11_sys.c	2008-05-01 10:46:23.000000000 -0400
@@ -76,6 +76,7 @@
 extern DEVICE rk_dev;
 extern DEVICE rl_dev;
 extern DEVICE hk_dev;
+extern DEVICE ke_dev;
 extern DEVICE rx_dev;
 extern DEVICE ry_dev;
 extern DEVICE mba_dev[];
@@ -129,6 +130,7 @@
     &rk_dev,
     &rl_dev,
     &hk_dev,
+    &ke_dev,
     &rx_dev,
     &ry_dev,
     &rp_dev,
